service/src/device_method.ts

```diff
@@ -63,7 +63,8 @@ export class DeviceMethod {
     /*Codes_SRS_NODE_IOTHUB_DEVICE_METHOD_16_008: [The `invokeOn` method shall throw a `ReferenceError` if `deviceId` is `null`, `undefined` or an empty string.]*/
     if (deviceId === null || deviceId === undefined || deviceId === '') throw new ReferenceError('deviceId cannot be \'' + deviceId + '\'');
 
-    const path = '/twins/' + deviceId + '/methods' + endpoint.versionQueryString();
+    /*Codes_SRS_NODE_IOTHUB_DEVICE_METHOD_16_017: [The `invokeOn` method shall uri-encode the device id.]*/
+    const path = '/twins/' + encodeURIComponent(deviceId) + '/methods' + endpoint.versionQueryString();
```

Comment:
> encodeURIComponent(deviceId) [](start = 28, length = 29)

Certainly the correct fix.

We have general problems with not uri encoding the device id.  It is certainly the reason that the previous bug is there.

I'm currently looking at the http sources to make sure those are correct and will look at the transports shortly.